File: ruby/opencv-video.rb
#!/usr/bin/env ruby
#
# Plays back a video in a window.

require "opencv"

def escape_pressed?(wait_msec)
  OpenCV::GUI::wait_key(wait_msec) == 27
end

def play_video(video_file, fps = 3)
  wait_delay = 100 / [1, fps].max

  window = OpenCV::GUI::Window.new('video')
  capture = OpenCV::CvCapture.open(video_file)

  while not escape_pressed?(wait_delay) and capture.grab
    frame = capture.retrieve
    break unless frame
    window.show(frame)
  end
end

video_file = ARGV.shift
abort "Usage: #{$PROGRAM_NAME} <video-file>" unless video_file

play_video(video_file)

File: python/opencv-video.py
#!/usr/bin/env python

import sys
import cv

def play(video_file):
    WINDOW_TITLE  = 'video'
    ESCAPE_KEY    = 27

    video  = cv.CaptureFromFile(video_file)
    fps    = cv.GetCaptureProperty(video, cv.CV_CAP_PROP_FPS)
    delay  = int(1000 / fps)

    while True:
        frame = cv.QueryFrame(video)
        if frame is None:
            break
        cv.ShowImage(WINDOW_TITLE, frame)
        if cv.WaitKey(delay) == ESCAPE_KEY:
            break

if __name__ == "__main__":
    play(sys.argv[1])

